apps/src/code-studio/pd/foorm/FoormEditorManager.jsx

```diff
@@ -0,0 +1,103 @@
+// Main page for Foorm Editor interface. Will initially show a choice
+// between loading an existing configuration or an empty configuration.
+// After that choice is made, will render FoormEditor with the chosen configuration.
+
+import React from 'react';
+import {connect} from 'react-redux';
+import PropTypes from 'prop-types';
+import {Button, DropdownButton, MenuItem} from 'react-bootstrap';
+import FoormEditor from './FoormEditor';
+
+class FoormEditorManager extends React.Component {
+  static propTypes = {
+    updateFormQuestions: PropTypes.func,
+    populateCodeMirror: PropTypes.func,
+    resetCodeMirror: PropTypes.func,
+    formNamesAndVersions: PropTypes.array,
+
+    // populated by redux
+    formQuestions: PropTypes.object
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      formKey: 0,
+      formPreviewQuestions: null,
+      formattedConfigurationOptions: this.getFormattedConfigurationDropdownOptions(
+        this.props.formNamesAndVersions
+      ),
+      showCodeMirror: false,
+      formName: null,
+      formVersion: null
+    };
+  }
+
+  getFormattedConfigurationDropdownOptions(formNamesAndVersions) {
+    return formNamesAndVersions.map((formNameAndVersion, i) => {
+      const formName = formNameAndVersion['name'];
+      const formVersion = formNameAndVersion['version'];
+      return (
+        <MenuItem
+          key={i}
+          eventKey={i}
+          onClick={() => this.loadConfiguration(formName, formVersion)}
+        >
+          {`${formName}, version ${formVersion}`}
+        </MenuItem>
+      );
+    });
+  }
+
+  loadConfiguration(formName, formVersion) {
+    $.ajax({
+      url: '/api/v1/pd/foorm/form_questions',
+      type: 'get',
+      data: {name: formName, version: formVersion}
+    }).done(result => {
+      this.props.updateFormQuestions(result);
+      this.setState({
+        showCodeMirror: true,
+        formName: formName,
+        formVersion: formVersion
+      });
+      this.props.resetCodeMirror(result);
+    });
+  }
+
+  initializeEmptyCodeMirror = () => {
+    this.setState({
+      showCodeMirror: true,
+      formName: null,
+      formVersion: null
+    });
+    this.props.resetCodeMirror({});
+  };
+
+  render() {
+    return (
+      <div>
+        <div>
+          <DropdownButton id="load_config" title="Load Survey...">
+            {this.state.formattedConfigurationOptions &&
```

Comment:
What happens if you don't repeat this?

Reply:
good question...turns out I don't need to repeat this. Thought it would have some weird behavior if it was null but it doesn't